src/uu/rm/src/rm.rs

```diff
@@ -87,17 +85,63 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         .map(|v| v.map(ToString::to_string).collect())
         .unwrap_or_default();
 
-    let force = matches.get_flag(OPT_FORCE);
+    let force_flag = matches.get_flag(OPT_FORCE);
 
-    if files.is_empty() && !force {
+    // If -f(--force) is before any -i (or variants) we want prompts else no prompts
+    let force_prompt_never: bool = if force_flag {
```

Comment:
Nice! I think the logic is perfect now! Those indices are incredibly weird in clap 4. Maybe we should open an issue for them on clap. It's so crazy that default values can have higher indices than user-supplied values.

I think we can make it a bit more terse, though. Maybe something like:
```rust
let force_prompt_never = force_flag && {
    let force_index = matches.index_of(OPT_FORCE).unwrap();
    [OPT_PROMPT, OPT_PROMPT_MORE, OPT_INTERACTIVE]
        .iter()
        .all(|opt| {
            matches.value_source(opt) != Some(ValueSource::CommandLine)
                || matches.index_of(opt).unwrap() < force_index
        })
};
```
If you don't like the unwraps, we could do `unwrap_or(0)` instead.

Thank you for diving deep into this!

Reply:
NP, thanks for helping so much throughout the process